ui/dashboards/src/context/DashboardProvider.tsx

```diff
@@ -26,8 +26,11 @@ interface DashboardState {
 interface DashboardActions {
   setEditMode: (isEditMode: boolean) => void;
   setLayouts: (layouts: LayoutDefinition[]) => void;
+  addLayout: (layout: LayoutDefinition) => void;
+  addItemToLayout: (index: number, item: GridItemDefinition) => void;
   setPanels: (panels: Record<string, PanelDefinition>) => void;
   addPanel: (name: string, panel: PanelDefinition) => void;
+  getDashboard: () => DashboardSpec;
```

Comment:
I'd second Christine's question about why this is a function rather than just a property on the state? I know you pulled layouts/panels out in your initial PR, but maybe the overall spec should just be in the state instead?